lib/hitters.go

```diff
@@ -0,0 +1,209 @@
+package vegeta
+
+import (
+	"bytes"
+	"crypto/tls"
+	"io"
+	"io/ioutil"
+	"math"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/valyala/fasthttp"
+)
+
+// TODO: Go-docs for everything here.
+// TODO: Overhaul examples in README to use new APIs
+// TODO: Fix attacker tests
+// TODO: Write CLI integration tests
+
+type Hitter interface {
```

Comment:
Unrelated to the (awesome) `fasthttp` addition, this abstraction would make it a lot easier to "re-purpose" Vegeta for SQL/general TCP load testing. A Postgres Hitter could (somewhat) easily be implemented on top of this. 

I realise I'd still have to create an interface for the `Target` struct, but it's at least one step closer.

Some very old prior art that is basically what I'd need: https://github.com/masahide/vegeta-mysql